libs/langchain-google-common/src/utils/gemini.ts

```diff
@@ -74,6 +83,29 @@ function messageContentImageUrl(
   }
 }
 
+function messageContentToMedia(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  content: Record<string, any>
+): GeminiPartInlineData | GeminiPartFileData {
+  if ("mimeType" in content && "data" in content) {
+    return {
+      inlineData: {
+        mimeType: content.mimeType,
+        data: content.data,
+      },
+    };
+  } else if ("mimeType" in content && "fileUri" in content) {
```

Comment:
Pondering out loud:
- perhaps instead of "fileUri" we just make it "uri" or "url"
- then we just use `extractMimeType` method as above to see if it's a data: url (and "mimeType" is optional or ignored) and use `inlineData`
- If it isn't a data url, and we have the mime type, then we pass the url and mime types given to us using `fileData`

This would then turn the `messageContentImageUrl` function into calling this function with the attributes of `content` set.

(And it also means that if we add more sophisticated file handling later, we only have to change it in one place.)